core/utils.py

```diff
@@ -370,32 +371,45 @@ def validate_json_schema(schema):
         ]
 
 
-def validate_password(password: str) -> None:
-    schema = PasswordValidator()
-    schema.min(settings.PASSWORD_MIN_LENGTH)
-    requirements = {
-        'PASSWORD_UPPERCASE': 'uppercase',
-        'PASSWORD_LOWERCASE': 'lowercase',
-        'PASSWORD_DIGITS': 'digits',
-        'PASSWORD_SYMBOLS': 'symbols'
-    }
-
-    for setting, method in requirements.items():
-        if getattr(settings, setting) > 0:
-            getattr(schema.has(), method)()
-
-    if not schema.validate(password):
-        raise GraphQLError(
-            f"Password must be at least {settings.PASSWORD_MIN_LENGTH} characters long, "
-            f"have at least {settings.PASSWORD_UPPERCASE} uppercase letter(s), "
+class CustomPasswordValidator:
+    def __init__(self):
+        self.schema = PasswordValidator()
+        self.requirements = {
+            'PASSWORD_UPPERCASE': 'uppercase',
+            'PASSWORD_LOWERCASE': 'lowercase',
+            'PASSWORD_DIGITS': 'digits',
+            'PASSWORD_SYMBOLS': 'symbols'
+        }
+        self.set_password_policy()
+
+    def set_password_policy(self):
+        self.schema.min(settings.PASSWORD_MIN_LENGTH)
+        for setting, method in self.requirements.items():
+            if getattr(settings, setting) > 0:
+                getattr(self.schema.has(), method)()
+
+    def validate(self, password, user=None):
+        if os.environ.get("MODE", "DEV") != "PROD":
+            if not self.schema.validate(password):
+                raise ValidationError(
+                    f"Password must be at least {settings.PASSWORD_MIN_LENGTH} characters long, "
+                    f"have at least {settings.PASSWORD_UPPERCASE} uppercase letter(s), "
+                    f"{settings.PASSWORD_LOWERCASE} lowercase letter(s), "
+                    f"{settings.PASSWORD_DIGITS} number(s), and "
+                    f"{settings.PASSWORD_SYMBOLS} special character(s)."
```

Comment:
that should avoid code duplications

```suggestion
                    self.get_help_text()
```